.cm/gitstream.cm

```diff
@@ -0,0 +1,48 @@
+# -*- mode: yaml -*-
+manifest:
+  version: 1.0
+
+automations:
+  # Enforces a certain format on PR titles
+  enforce_pr_title:
+    if:
+      - {{ pr.title | match(regex=titlePolicy.titleRegex) | nope }}
+    run:
+      - action: request-changes@v1
+        args:
+          comment: |
+            All PRs must be titled according to our semantic naming policy: `[primary-label] <short summary>`
+  # Add a label that indicates how many minutes it will take to review the PR.
+  estimated_time_to_review:
+    if:
+      - true
+    run:
+      - action: add-label@v1
+        args:
+          label: "{{ calc.etr }} min review"
+          color: {{ colors.error if (calc.etr >= 20) else (colors.warning if (calc.etr >= 5) else colors.success) }}
+  # Post a comment that lists the best experts for the files that were modified.
+  explain_code_experts:
+    if:
+      - { { pr.labels | match(term='Auto assign') | some } }
```

Comment:
Do you know if the reviewer would be assigned only from within the organization?
Most of our community contributions are for localization. Imagine a case, where some community member introduces a new locale (translations), and later, someone amends it.
Would this tool would probably assign the author of the file as the reviewer? 🤔 

It's not a blocker as this is an opt-in option, but just an idea if we ever want to enable this by default. 🤔 

P.S. Correct me if I'm wrong, but would this action assign a reviewer? In one place it says that it does, in another, there is no mention of it... 🤷

Reply:
Is there a way to configure load balancing of requests of reviews? I am afraid that we may end up with requesting review from only 2-3 team members (the ones that are working on the repository longest). Or have a max number of request of review someone can have at certain point.

Reply:
That's a great question and I definitely do not have an answer to that as of now. I will dig the docs to find out and contact the gitStream devs 👍🏼

Reply:
As I understand gitStream, it doesn't support the auto-assign logic for issues? We might need to developer something custom for issues at which point we could reapply it with PRs?

Reply:
There are lots of great questions in this thread, I'll try to answer all of them.

> Do you know if the reviewer would be assigned only from within the organization?

I will double-check on this, but I'm not sure if this is possible yet. If not, we may be able to prioritize adding this capability.

> Would this tool would probably assign the author of the file as the reviewer?

Generally speaking, yes, if you use the `codeExperts` Filter function. This returns up to 2 potential reviewers based on past activity, excluding the PR author. However, you can configure gitStream to treat specific resources differently. For example, you could configure a different set of rules for translation resources that assign a specific team or individual.

> P.S. Correct me if I'm wrong, but would this action assign a reviewer? In one place it says that it does, in another, there is no mention of it...

This is correct, the current config we have here wouldn't assign a reviewer, but that's just a matter of using the [`add-reviewers`](https://docs.gitstream.cm/automation-actions/#add-reviewers) automation action.

> Is there a way to configure load balancing of requests of reviews?

There are a few ways to handle this:

1. you can assign a team to review the PR and rely on GitHub's built-in configuration to determine who gets assigned (there are options for round robin and load balance).
1. You can use [intersection](https://docs.gitstream.cm/filter-functions/#intersection) to compare the list returned from Code Experts to a pre-defined list of reviewers. You could also set a maximum threshold for Code Experts to avoid assigning reviews to people who mostly contribute new code.
1. You can use the nunjucks `random` function to assign a reviewer from a pre-defined list.

> As I understand gitStream, it doesn't support the auto-assign logic for issues?

Correct, gitStream lacks any integrations with GitHub Issues. Its something we expect to add in the future, but have been waiting for an excuse to prioritize it. If you can share more details about this use case we can figure out what it would take to put this on the roadmap.